pkg/process/checks/allprocesses_windows.go

```diff
@@ -91,6 +95,9 @@ func getAllProcStats(probe *procutil.Probe, pids []int32) (map[int32]*procutil.S
 }
 
 func getAllProcesses(probe *procutil.Probe) (map[int32]*procutil.Process, error) {
+	allProcessesMutex.Lock()
+	defer allProcessesMutex.Unlock()
```

Comment:
Do we know how long this function takes? I worry this largely scoped mutex might have functionality impacts.

Reply:
Short answer, no, although the full check runs generally appear (in the logs) to take around 10ms (plus or minus, of course, depending on number of processes).